### DISCUSSION TITLE
Digital Crossover - Bi-Amping using snapcast

### DISCUSSION THREAD

**Discussion by giggywithit** [snapcast/snapcast#767] [Ideas]
Hello

Reading through the discussion board here it doesn't look like a busy project or actively discussed at least.
I am wondering what the possibility would be to incorporate some filters into the process for use in bi-amping multiple separate clients into projects such as subwoofers and channeling mid and high frequencies. Are there any known incorporations of this idea in use today? Whats the possibility of such and timeframe?

Thanks
S

**Comment by mrwsl** [comment]
> Reading through the discussion board here it doesn't look like a busy project or actively discussed at least.

It's an indicator on how well this project works! No one has questions and is happy ;)

> ... for use in bi-amping  ...

I'm curious: isn't that a hardware thing? How would it connect to a project like snapcast?

**Reply by badaix** [reply]
> It's an indicator on how well this project works! No one has questions and is happy ;)

For me it's the first time that I visit the discussion board after creating it some weeks ago. I don't get any notifications, which is ok, because I'm busy enough caring about Issues. Happy and grateful to see people responding to and helping other users.

**Comment by giggywithit** [comment]
Hi and thank you for the response

Filtering can be done either way, hardware or software, but software allows a much greater control and configuration and less cost. Instead of having to rewire an amp for specific individual uses, one could change a few lines of config file. I may tinker around with attempting to get some ALSA filters working on the client side of things. This would be a game changer for my outfit.

Any ideas why I am getting this error?
`sudo cat /dev/urandom > /tmp/snapfifo`
`bash: /tmp/snapfifo: Permission denied`

I cannot seem to get any audio on the client side although it apparently sees (is connected) the server.
I also tried VLC on the server side machine and setting it's output to 'file' and giving it the same destination as the server below..
`source = file:///home/cox/audiofile/snapaudiofile`
No audio

**Reply by gribouk** [reply]
As I understood, you want to take a signal from a single pipe, separate frequencies and direct the result to different speakers dedicated to pay these specific frequencies?
If so, let me offer you a solution, which I use:
1. Run as many snapclients on your machine as you have speakers attached to it and let them read from same pipe.
2. For every speaker use software mixer  to cut the unwanted spectrum.
Done!

Actually I use it otherway around, I have several pipes, several snapclients for each pipe and they play to one speaker (set of speakers actually) - this way I mix TTS and music without interrupting music. I just lower the output of snapclient for the music when TTS comes, and otherway aroung when it stops...

**Reply by giggywithit** [reply]
That sounds like a cool way of being a DJ

I am still interested in trying out the Bi-Amping at some point, however, my resources are being used in many other places at the moment. I will revisit the idea within the year maybe.
Thanks

**Comment by mrwsl** [comment]
> Any ideas why I am getting this error?

Try `sudo sysctl fs.protected_fifos=0`

EDIT: Background: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/sysctl/fs.txt?id=30aba6656f61ed44cba445a3c0d38b296fa9e8f5
